tests/unit_tests/shared/share/test_shell.py

```diff
@@ -283,6 +283,51 @@ def test_that_delete_directory_does_not_follow_symlinks(shell):
     assert os.path.exists("link_target/link_file")
 
 
+@pytest.mark.usefixtures("use_tmpdir")
+def test_that_delete_directory_on_a_symlink_to_file_is_ignored(shell):
+    with open("link_target", "w", encoding="utf-8") as f:
+        f.write("hei")
+    shell.symlink("link_target", "link")
+    shell.delete_directory("link")
+    assert os.path.exists("link_target")
+    assert os.path.exists("link")
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+def test_that_delete_directory_on_a_symlink_to_a_directory_only_deletes_link(shell):
+    shell.mkdir("link_target")
+    with open("link_target/file", "w", encoding="utf-8") as f:
+        f.write("hei")
+    shell.symlink("link_target", "link")
+    shell.delete_directory("link")
+    assert os.path.exists("link_target/file")
+    assert not os.path.exists("link")
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+@pytest.mark.parametrize("trailing", ["", "/"])
+def test_that_delete_directory_on_a_symlink_to_a_directory_is_conditionally_ignored(
+    shell, trailing
+):
+    """A trailing slash turns the world upside down"""
```

Comment:
:laughing: the comment could be more descriptive - maybe to detailed down both scenarios

Reply:
Written more now